packages/wonder-blocks-tooltip/components/tooltip-tail.js

```diff
@@ -171,38 +166,58 @@ export default class TooltipArrow extends React.Component<Props> {
             return null;
         }
         const {placement} = this.props;
-        const {height, y, width, x} = position;
-        // TODO(somewhatabstract): Use unique IDs!
+        const {y, x, offsetShadowX} = position;
         const dropShadowFilterId = `tooltip-dropshadow-${placement}-${tempIdCounter++}`;
         return [
             <filter
                 key="filter"
                 id={dropShadowFilterId}
-                width={width}
+                // Height and width tell the filter how big of a canvas to
+                // draw based on its parent size. i.e. 2 times bigger.
+                // This is so that the diffuse gaussian blur has space to
+                // bleed into.
+                width={"200%"}
+                height={"200%"}
+                // The x and y values tell the filter where, relative to its
+                // parent, it should begin showing its canvas. Without these
+                // the filter would clip at 0,0, which would look really
+                // strange.
                 x={x}
-                height={height}
                 y={y}
             >
+                {/* Here we provide a nice blur that will be our shadow
+                * The stdDeviation is the spread of the blur. We don't want it
+                * too diffuse.
+                */}
                 <feGaussianBlur
                     in="SourceAlpha"
-                    stdDeviation={Spacing.xSmall / 2}
+                    stdDeviation={Spacing.xxSmall / 2}
                 />
-                <feMerge>
-                    <feMergeNode />
-                    <feMergeNode in="SourceGraphic" />
-                </feMerge>
+
+                {/* Here we adjust the alpha (feFuncA) linearly so as to blend
+                * the shadow to match the rest of the tooltip bubble shadow.
+                * It is a combination of the diffuse blur and this alpha
+                * value that will make it look right.
+                */}
+                <feComponentTransfer>
+                    <feFuncA type="linear" slope="0.3" />
+                </feComponentTransfer>
             </filter>,
             /**
              * Draw the tooltip arrow and apply the blur filter we created
              * above, to produce a drop shadow effect.
+             * We move it down a bit with a translation, so that it is what
+             * we want.
              */
-            <polyline
-                key="dropshadow"
-                fill={Colors.offBlack16}
-                points={points.join(" ")}
-                stroke={Colors.offBlack32}
-                filter={`url(#${dropShadowFilterId})`}
-            />,
+            <g transform={`translate(${offsetShadowX},5.5)`}>
```

Comment:
Why do we need the `offsetShadowX`?

Reply:
Because we are pulling the arrow left or right by a pixel in order to make sure the triangle is flush to the bubble border. We don't want the shadow to overdraw that border, so it needs to adjust for that 1px.

Reply:
Added commentary